flink-core/src/main/java/org/apache/flink/util/MdcUtils.java

```diff
@@ -45,7 +45,13 @@ public class MdcUtils {
     public static MdcCloseable withContext(Map<String, String> context) {
         final Map<String, String> orig = MDC.getCopyOfContextMap();
         MDC.setContextMap(context);
-        return () -> MDC.setContextMap(orig);
+        return () -> {
```

Comment:
can we have test for this please.

Reply:
How do you propose we test this?

MDC used in tests will never result in `orig` being `null`. Should I add a separate Maven project with Logback for this?

Reply:
I added a simple mocked test so that we can see how it looks